Adyen/ApiSerialization/Converter/JsonConvertSerializerWrapper.cs

```diff
@@ -11,19 +13,31 @@ internal class JsonConvertSerializerWrapper
 
         internal static string Serialize(SaleToPOIMessage saleToPoiMessage)
         {
-            var serialize= JsonConvert.SerializeObject(saleToPoiMessage,
-                new SaleToPoiMessageConverter(),
-                new StringEnumConverter(),
-                new IsoDateTimeConverter { DateTimeFormat = DateTimeFormat });
+            var serialize = JsonConvert.SerializeObject(saleToPoiMessage,
+                GetSerializerSettings(new SaleToPoiMessageConverter()));
             return serialize;
         }
 
         internal static string Serialize(SaleToPoiMessageSecured saleToPoiMessageSecured)
         {
             return JsonConvert.SerializeObject(saleToPoiMessageSecured,
-                                               new SaleToPoiMessageSecuredConverter(),
-                                               new StringEnumConverter(),
-                                               new IsoDateTimeConverter { DateTimeFormat = DateTimeFormat });
+                GetSerializerSettings(new SaleToPoiMessageSecuredConverter()));
+        }
+
+        private static JsonSerializerSettings GetSerializerSettings(JsonConverter messageConverter)
```

Comment:
nit: Let's group everything in one function and remove the `GetSerializerSettings(...)` function. When navigating through the library, it only takes one jump to understand how the `JsonSerializerSettings` is constructed

Reply:
Does this work?
Or should we inline method and duplicate code which creates `JsonSerializerSettings`?

Reply:
I'd consider initializing the JsonSerializerSettings once, as now it's created every function-call

Reply:
Two private static fields with settings for `SaleToPOIMessage` and for `SaleToPoiMessageSecured`?

Reply:
Works for me!

Reply:
I'm afraid I haven't got an idea.

When `CreateSerializerSettings` was initially defined it created `JsonSerializerSettings` each time it is called. `CreateSerializerSettings` was called from both `Serialize` methods. So, `JsonSerializerSettings` was instantiated per each `Serialize` call (the same as was before for converters). We could move it to both `Serialize` methods but nothing would change in such case from settings instantiation standpoint (but would require to change configuration if two places if necessary). Also, we can make settings static not to instantiate it per each call which is done in current implementation. Is there any other approach? Am I missing something?

Could you explain more clearly which solution you suggest, please?

Reply:
I was referring to: We could pass the `JsonSerializerSettings` in the parameters of the `Serialize(SaleToPoiMessageSecured saleToPoiMessageSecured, JsonSerializerSettings ...)` and `Serialize(SaleToPOIMessage saleToPoiMessage, JsonSerializerSettings ...)` methods. This allows consumers of the library to pass their (custom) `JsonSerializerSettings` if needed.

I like your current approach, this will do fine.